### ISSUE TITLE
pipeline init should be flexible with the repo URL

### ISSUE THREAD

**Issue by kohidave** [aws/copilot-cli#1153]
When customers run `pipeline init` - if they walk through the wizard, we'll correctly extract the repo URL (yay!). 

When customers provide their own URL, they may include the optional `.git` at the end of the URL: 

Works:
```sh

  $ copilot pipeline init \
   --github-url https://github.com/gitHubUserName/myFrontendApp \
   --github-access-token file://myGitHubToken \
   --environments "stage,prod" \
   --deploy

```

Doesn't work:
```sh

  $ copilot pipeline init \
   --github-url https://github.com/gitHubUserName/myFrontendApp.git \
   --github-access-token file://myGitHubToken \
   --environments "stage,prod" \
   --deploy

```

Pipeline init should clean up the GH URL if we detect the customer has passed in a format that we're not expecting.

**Comment by git823** [comment]
There are two problems. The repository URL with ".git" suffix is used for CFn stack name and CodePipeline's configuration property. The name of CFn stack is not allowed to contain dots. Also, Repo property for GitHub source action on CodePipeline needs to be without ".git" suffix. I'll fix it and send PR.

**Comment by efekarakus** [comment]
Resolved in #1215
